modules/gsp-namespace-role/iam.tf

```diff
@@ -0,0 +1,39 @@
+data "aws_iam_policy_document" "assume-role" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals = {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${var.account_id}:role/${var.cluster_name}_kiam_server"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "namespace-defaults" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "sqs:SendMessage",
+      "sqs:ReceiveMessage"
+    ]
+
+    resources = ["arn:aws:sqs:*:${var.account_id}:${var.cluster_name}-sqsqueue-${var.namespace_name}-*"]
+  }
+}
+
+resource "aws_iam_role" "namespace" {
+  name               = "${var.cluster_name}-namespace-${var.namespace_name}"
```

Comment:
is it worth considering a `path` here? https://www.terraform.io/docs/providers/aws/r/iam_role.html#path